cfme/tests/infrastructure/test_host.py

```diff
@@ -349,13 +351,27 @@ def test_hosts_not_displayed_several_times(appliance, provider, setup_provider):
     assert host_count == navigate_to(appliance.collections.hosts, "All").paginator.items_amount
 
 
-@pytest.fixture
-def setup_provider_min_hosts(request, provider, min_hosts=2):
-    num_hosts = len(provider.data.get('hosts', {}))
-    if num_hosts < min_hosts:
-        pytest.skip(f'Not enough hosts({num_hosts}) to run test. Need at least {min_hosts}')
+@pytest.fixture(params=[1, 2, 'all'])
+def setup_provider_min_hosts(request, provider):
+    if request.param == 1 and "compare" in request._pyfuncitem.name:
+        pytest.skip(f'Compare tests need at least 2 hosts.')
+    elif request.param == "all":
+        min_hosts = len(provider.data.get('hosts', {}))
+        if min_hosts <= 2:
+            pytest.skip(f'"All" hosts case is covered by previous test for {min_hosts} Hosts.')
+        request.param = min_hosts
+    else:
+        min_hosts = request.param
+        num_hosts = len(provider.data.get('hosts', {}))
+        if num_hosts < min_hosts:
+            pytest.skip(f'Not enough hosts({num_hosts}) to run test. Need at least {min_hosts}')
```

Comment:
I would update the message here to better differentiate where the limitation came from. Something like `f'Number of hosts defined in yaml for {provider} does not meet minimum for test parameter {min_hosts}'`